code/modules/overmap/ships/controlled_ship_datum.dm

```diff
@@ -0,0 +1,236 @@
+	if(istype(dock_requester, /datum/overmap/ship/controlled))
+		var/datum/overmap/ship/controlled/docker_port = dock_requester
+		shuttle_port.shuttle_areas += docker_port.shuttle_port.shuttle_areas
+
+/**
+  * Docks to an empty dynamic encounter. Used for intership interaction, structural modifications, and such
+  */
+/datum/overmap/ship/controlled/proc/dock_in_empty_space()
+	var/datum/overmap/dynamic/empty/E = locate() in SSovermap.overmap_container[x][y]
+	if(!E)
+		E = new(list("x" = x, "y" = y))
+	if(E) //Don't make this an else
+		Dock(E)
+
+/datum/overmap/ship/controlled/burn_engines(n_dir = null, percentage = 100)
+	if(docked_to || docking)
+		CRASH("[src] burned engines while docking or docked!")
+
+	var/thrust_used = 0 //The amount of thrust that the engines will provide with one burn
+	refresh_engines()
+	if(!mass)
+		calculate_mass()
+	calculate_avg_fuel()
+	for(var/obj/machinery/power/shuttle/engine/E as anything in shuttle_port.engine_list)
+		if(!E.enabled)
+			continue
+		thrust_used += E.burn_engine(percentage)
+	est_thrust = thrust_used //cheeky way of rechecking the thrust, check it every time it's used
+	thrust_used = thrust_used / max(mass * 100, 1) //do not know why this minimum check is here, but I clearly ran into an issue here before
+	if(n_dir)
+		accelerate(n_dir, thrust_used)
+	else
+		decelerate(thrust_used)
+
+/**
+  * Just double checks all the engines on the shuttle
+  */
+/datum/overmap/ship/controlled/proc/refresh_engines()
+	var/calculated_thrust
+	for(var/obj/machinery/power/shuttle/engine/E as anything in shuttle_port.engine_list)
+		E.update_engine()
+		if(E.enabled)
+			calculated_thrust += E.thrust
+	est_thrust = calculated_thrust
+
+/**
+  * Calculates the mass based on the amount of turfs in the shuttle's areas
+  */
+/datum/overmap/ship/controlled/proc/calculate_mass()
+	. = 0
+	var/list/areas = shuttle_port.shuttle_areas
+	for(var/shuttle_area in areas)
+		. += length(get_area_turfs(shuttle_area))
+	mass = .
+
+/**
+  * Calculates the average fuel fullness of all engines.
+  */
+/datum/overmap/ship/controlled/proc/calculate_avg_fuel()
+	var/fuel_avg = 0
+	var/engine_amnt = 0
+	for(var/obj/machinery/power/shuttle/engine/E as anything in shuttle_port.engine_list)
+		if(!E.enabled)
+			continue
+		fuel_avg += E.return_fuel() / E.return_fuel_cap()
+		engine_amnt++
+	if(!engine_amnt || !fuel_avg)
+		avg_fuel_amnt = 0
+		return
+	avg_fuel_amnt = round(fuel_avg / engine_amnt * 100)
+
+/datum/overmap/ship/controlled/tick_move()
+	if(avg_fuel_amnt < 1)
+		decelerate(max_speed / 100)
+	..()
+
+/**
+  * Bastardized version of GLOB.manifest.manifest_inject, but used per ship
+  *
+  * * H - Human mob to add to the manifest
+  * * C - client of the mob to add to the manifest
+  * * human_job - Job of the human mob to add to the manifest
+  */
+/datum/overmap/ship/controlled/proc/manifest_inject(mob/living/carbon/human/H, client/C, datum/job/human_job)
+	set waitfor = FALSE
```

Comment:
waitfor is FALSE by default as of a few years ago iirc

Reply:
Just a leftover from manifest_inject(), the original proc, removing